docs/content/en/docs/concepts/plugins.md

```diff
@@ -0,0 +1,132 @@
+
+This page serves as a guide for understanding and developing Reva plugins.
+
+### How Plugins Work
+
+A Reva plugin represents a *service driver*.
+
+Reva plugins are completely seperate, standalone applications that the core of Reva starts and communicates with. These plugin applications aren't meant to be run manually. Instead, Reva launches and communicates with them. 
+
+Reva uses hashicorp's [go-plugin](https://github.com/hashicorp/go-plugin) framework to implement the plugin architecture. The plugin processes communicate with the Core using the go-plugin framework, which internally uses RPCs for the communication. The Reva core itself is responsible for launching and cleaning up the plugin processes.
+
+### Developing Plugins
+
+Reva plugins must be written in [Go](https://golang.org/), so you should be familiar with the language.
+
+As mentioned earlier, the components that can be created and used in a Reva Plugin are service drivers. These drivers can belong to any of the Reva service, eg: Userprovider, Storageprovider, Authprovider etc. Each service exposes an [interface](https://golang.org/doc/effective_go#interfaces_and_types) for the plugin to implement.
+
+All you need to do to create a plugin is:
+
+1. Create an implementation of the desired interface: A Plugin should implement the interface exposed by the corresponding service. Eg: [Userprovider](https://github.com/cs3org/reva/blob/master/pkg/user/user.go#L67) interface.
+2. Serve the plugin using the [go-plugin's](https://github.com/hashicorp/go-plugin) `plugin.Serve` method.
+
+The core handles all of the communication details and go-plugin implementations inside the server.
+
+Your plugin must use the packages from the Reva core to implement the interfaces. You're encouraged to use whatever other packages you want in your plugin implementation. Because plugins are their own processes, there is no danger of colliding dependencies.
+
+- `github.com/cs3org/reva/pkg/<service_type>`: Contains the interface that you have to implement for any give plugin.
+- `github.com/hashicorp/go-plugin`: To serve the plugin over RPC. This handles all the inter-process communication.
+
+Basic example of serving your component is shown below. This example consists of a simple `JSON` plugin driver for the [Userprovider](https://github.com/cs3org/reva/blob/master/internal/grpc/services/userprovider/userprovider.go) service. You can find the example code [here](https://github.com/cs3org/reva/blob/master/examples/plugin/json/json.go).
+
+```go
+
+// main.go
+
+import (
+   	"github.com/cs3org/reva/pkg/user"
+    "github.com/hashicorp/go-plugin"
+)
+
+// Assume this implements the user.Manager interface
+type Manager struct{}
+
+// Handshake hashicorp go-plugin handshake
+var Handshake = plugin.HandshakeConfig{
+	ProtocolVersion:  1,
+	MagicCookieKey:   "BASIC_PLUGIN",
```

Comment:
@jimil749 would make sense that the protocol version and Magic* contents are constants defined in the Reva plugin package that can be used?

Reply:
Yes, that would be in-fact better alternative. The developer can then just import the package and use the exported consts. :+1:

Reply:
Awesome

Reply:
I'll update the documentation once we get #1970 merged. (There might be more changes)